### PR TITLE
show appropriate message and inhibit motion when CAEN PVs are unavail…

### PR DESCRIPTION
**Pull request by RaiBishal**
# Description of work

Sans2d vacuum tank to have an indicator when CAEN power supplies are on.

### Ticket

https://github.com/ISISComputingGroup/IBEX/issues/5626

### Acceptance criteria

- CAEN power supply LED indicator turns on when appropriate and motion is inhibited
- When unable to read from CAENs, motion is inhibited 

Note:- requires latest SANS2D config `custom_records.db`
### Unit tests

*Give an overview of unit tests you have added or modified, if applicable. The aim is provide information to help the reviewer*

### System tests

*Mention any automated tests or manual tests that you have added or modified, if applicable. The aim is provide information to help the reviewer*

### Documentation
*Highlight and provide a link to any additions or changes to the documentation, if applicable. The aim is provide information to help the reviewer*

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?
- [ ] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### Final Steps
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

**Review by JamesKingWork — DISMISSED** [review]
See comments

File: base/uk.ac.stfc.isis.ibex.opis/resources/sans2d_vacuum_tank.opi [hunk 1]
```diff
@@ -1945,6 +1945,116 @@ $(pv_value)</tooltip>
...
+          <enabled>true</enabled>
+          <font>
+            <opifont.name fontName="Segoe UI" height="9" style="0" pixels="false">Default</opifont.name>
+          </font>
+          <forecolor_alarm_sensitive>false</forecolor_alarm_sensitive>
+          <foreground_color>
+            <color red="192" green="192" blue="192" />
+          </foreground_color>
+          <height>25</height>
+          <name>LED</name>
```
[the remark this answers is not in the document]
Comment (RaiBishal) [reply base/uk.ac.stfc.isis.ibex.opis/resources/sans2d_vacuum_tank.opi:1977]: If invalid and not ON, there will be a purple border around the LED it in both advanced and user tabs now.

File: base/uk.ac.stfc.isis.ibex.opis/resources/sans2d_vacuum_tank.opi [hunk 2]
```diff
@@ -5028,16 +5643,25 @@ $(pv_value)</tooltip>
...
+            <exp bool_exp="pvInt0 == 2">
+              <value>true</value>
+            </exp>
             <pv trig="true">$(P)FRONTDET:ON</pv>
           </rule>
+          <rule name="Rule" prop_id="text" out_exp="true">
+            <exp bool_exp="pv0 == 2">
+              <value>"Unable to read from CAEN, front motion inhibited"</value>
+            </exp>
+            <pv trig="true">$(P)REARDET:ON</pv>
```
[the remark this answers is not in the document]
Comment (RaiBishal) [reply base/uk.ac.stfc.isis.ibex.opis/resources/sans2d_vacuum_tank.opi]: Changed.

File: base/uk.ac.stfc.isis.ibex.opis/resources/sans2d_vacuum_tank.opi [hunk 3]
```diff
@@ -5084,6 +5708,15 @@ $(pv_value)</tooltip>
...
               <value>true</value>
             </exp>
+            <exp bool_exp="pvInt0 == 2">
+              <value>true</value>
+            </exp>
+            <pv trig="true">$(P)REARDET:ON</pv>
+          </rule>
+          <rule name="Rule" prop_id="text" out_exp="true">
+            <exp bool_exp="pv0 == 2">
+              <value>"Unable to read from CAEN, rear motion inhibited"</value>
```
[the remark this answers is not in the document]
Comment (RaiBishal) [reply base/uk.ac.stfc.isis.ibex.opis/resources/sans2d_vacuum_tank.opi]: Done.

File: base/uk.ac.stfc.isis.ibex.opis/resources/sans2d_vacuum_tank.opi [hunk 4]
```diff
@@ -5028,16 +5643,25 @@ $(pv_value)</tooltip>
...
               <value>true</value>
             </exp>
+            <exp bool_exp="pvInt0 == 2">
+              <value>true</value>
+            </exp>
             <pv trig="true">$(P)FRONTDET:ON</pv>
           </rule>
+          <rule name="Rule" prop_id="text" out_exp="true">
+            <exp bool_exp="pv0 == 2">
+              <value>"Unable to read from CAEN, front motion inhibited"</value>
```
[the remark this answers is not in the document]
Comment (RaiBishal) [reply base/uk.ac.stfc.isis.ibex.opis/resources/sans2d_vacuum_tank.opi]: Done.

File: base/uk.ac.stfc.isis.ibex.opis/resources/sans2d_vacuum_tank.opi [hunk 5]
```diff
@@ -2340,6 +2450,511 @@ $(pv_value)</tooltip>
           <x>0</x>
           <y>72</y>
         </widget>
+        <widget typeId="org.csstudio.opibuilder.widgets.groupingContainer" version="1.0.0">
```
[the remark this answers is not in the document]
Comment (RaiBishal) [reply base/uk.ac.stfc.isis.ibex.opis/resources/sans2d_vacuum_tank.opi]: Removed the one I added

**Comment by RaiBishal** [comment]
Made the latest changes to `inhibitr.cmd` and `custom_records.db` in SANS2D config area

File: base/uk.ac.stfc.isis.ibex.opis/resources/sans2d_vacuum_tank.opi (hunk 1 above)
[the remark this answers is not in the document]
Comment (JamesKingWork) [reply base/uk.ac.stfc.isis.ibex.opis/resources/sans2d_vacuum_tank.opi:1977]: This is a really good solution, I like it

**Review by JamesKingWork — APPROVED** [review]
Looks really good and intuitive

**Pull request merged**